src/hooks/useHailFrequency.js

```diff
@@ -25,38 +26,37 @@ async function fetchFromHailFrequency({
   }).then((res) => res.json())
 }
 
-async function fetcher(query, spaceId, spaceToken) {
+async function fetcher(query, spaceId, spaceToken, endpoint, type) {
   let data = []
   let nextToken = ''
-  let page = 1
-
-  while (page === 1 || nextToken) {
+  // fetch the data as long as there's more
+  // use a do_while loop since we need to fetch at least 1 time
+  do {
     const res = await fetchFromHailFrequency({
       query,
       first: 1000,
-      nextToken,
       spaceId,
-      spaceToken
+      spaceToken,
+      nextToken,
+      endpoint
     })
 
-    if (res && res.data) {
-      const queryName = Object.keys(res.data).shift() // e.g. 'getProducts'
-      const queryResults = res.data[queryName]
+    let queryResults = res?.data?.[type]
```

Comment:
Using an explicit key/type feels a bit better than trying to hack it with Object.keys. But I'm open to other ideas!

Reply:
Why is `type` in square brackets here?

Reply:
It's sort of a safe way of doing this in one line: 
```js
let data = res?.data;
data[type]
```
You can use `?.["key"]` to optional chain string-based access (also works for array indexing!)

Reply:
Gotcha! Sorry for some reason I was thinking that `type` was a property on `res.data` - makes perfect sense now.